### DISCUSSION TITLE
Stacks with single vs multiple dependencies

### DISCUSSION THREAD

**Discussion by james64** [terramate-io/terramate#1793] [General]
Hey,

there are two ways how to define dependencies between stacks (ie db stack depends on vpc):
1. Nesting stacks in folder structure as defined [here](https://terramate.io/docs/cli/stacks/nesting). It seems to be presented as a primary way of encoding dependencies between stacks.
2. [Explicit configuration](https://terramate.io/docs/cli/stacks/configuration#explicit-order-of-execution) via after and before. This seems to be presented more as an escape hatch option rather then go-to solution.

Is my primary vs escape-hatch feeling correct? Is this how terramate creators think about their usage?

Nesting stacks can only express single dependency. I wonder how limiting this is in real world? How rare it is to have stack with multiple dependencies?

**Comment by i4ki** [comment]
Hi @james64 

Both cases happen in real world and in my experience it depends on the use case. Sometimes nesting solves but if the involved stacks do not belong to the same scope, then maybe makes more sense to organize them in their proper directories and use explicit after/before clauses. There are many cases where nesting works like a charm, for example `vpc-abc` and `vpc-abc/subnet-xyz` stacks, but other cases it does not.
It really depends... Terramate gives the two options and we are not opinionated about which approach you should use.

I hope that helps.

**Comment by james64** [comment]
Yes, this is helpful. Thanks for clarification :+1:
